src/bikeracks/votes.py

```diff
@@ -111,5 +111,53 @@ def submit_vote():
     # maybe return 500 here
     return "", 500
     
-
+@votes.route('/unvote', methods=['POST'])
+def unvote():
+    # remove a vote that the user had previously made
+    if request.method == 'POST':
+        try:
+            rack_id = request.args.get('rack_id') or None
+            user_id = request.args.get('user_id') or None
+            vote_type = request.args.get('vote_type') or None
+            
+            # connect to db
+            db = get_db()
+            
+            # remove user's vote for this rack from votes table and update bikeracks vote count
+            
+            #remove user's vote 
+            query = """
+                    DELETE FROM
+                        votes
+                    WHERE
+                        user_id=? AND
+                        vote_type=? AND
+                        rack_id=?
+                    """
+            db.execute(query, (user_id, vote_type, rack_id))
+            
+            # decrement the vote count in the bikeracks row for this rack
+            if int(vote_type) == 1:
+                vote_count = "upvote_count"
+            else:
+                vote_count = "downvote_count"
+            
+            query = """
+                        UPDATE
+                            bikeracks
+                        SET
+                            {}={} - 1
+                        WHERE
+                            rack_id=?
+                    """.format(vote_count, vote_count)
+            
+            db.execute(query, (rack_id,))
+            
+            db.commit()
+            return "OK", 200
+        except sqlite3.Error as e:
+            print("Database error:", e)
+            return "", 500
+        except KeyError as key_e:
+            print("Error with key: {}".format(key_e))
```

Comment:
What are you going to return to the client on `KeyError` ?

Reply:
500 response code

Reply:
An unhandled exception does that already, doesn't it? Why catch them here?

Also don't use `print` for actual logging. Use the `logging` module so that it goes to stderr instead of stdout, also it will have a timestamp associated with it and tell you what line it was on, etc.